ID.me WebVerify SDK/IDmeWebVerify.m

```diff
@@ -364,31 +406,50 @@ - (BOOL)application:(UIApplication *)application
+            self.codeVerifier = nil;
+            self.codeChallenge = nil;
+            self.codeChallengeMethod = nil;
+
+            NSString *params = [NSString stringWithFormat:@"client_id=%@&client_secret=%@&redirect_uri=%@&code=%@&grant_type=authorization_code&code_verifier=%@",
+                                self.clientID, self.clientSecret, self.redirectURI, code, codeVerifier];
 
             __weak IDmeWebVerify *weakSelf = self;
             [self makePostRequestWithUrl:[self urlStringWithQueryString:IDME_WEB_VERIFY_REFRESH_CODE_URL]
                               parameters:params
                               completion:^(NSData * _Nullable data, NSURLResponse * _Nullable response, NSError * _Nullable error) {
-                                      NSError *jsonError;
-                                      NSDictionary *json = [NSJSONSerialization JSONObjectWithData:data
-                                                                                           options:NSJSONReadingMutableContainers
-                                                                                             error:&jsonError];
-                                      if (json) {
-                                          [weakSelf saveTokenDataFromJson:json scope:requestScope];
-                                          [weakSelf callWebVerificationResultsWithToken:[json objectForKey:IDME_WEB_VERIFY_ACCESS_TOKEN_PARAM] error:nil];
-                                      } else {
-                                          [weakSelf callWebVerificationResultsWithToken:nil
-                                                                                  error:[weakSelf notAuthorizedErrorWithUserInfo:@{
-                                                                                            NSLocalizedDescriptionKey: IDME_WEB_VERIFY_VERIFICATION_FAILED
-                                                                                        }]];
-                                      }
-                                  }];
+                                  NSError *jsonError;
+                                  NSDictionary *json = [NSJSONSerialization JSONObjectWithData:data
+                                                                                       options:NSJSONReadingMutableContainers
+                                                                                         error:&jsonError];
+
+                                  NSInteger statusCode = [(NSHTTPURLResponse *) response statusCode];
+                                  if (error || statusCode < 200 || statusCode >= 300) {
+                                      // Error from server, we may have a response in the json
+                                      NSError *callbackError = [weakSelf codeAuthenticationErrorWithUserInfo:json];
+                                      [weakSelf callWebVerificationResultsWithToken:nil error:callbackError];
```

Comment:
I think there might be a race condition in that `_safariViewController` is set to nil before calling this and therefore a new request might have overriden the callback. Maybe we need a boolean flag to signal if a new request can open the SafariViewController and not check for it to be nil.